### ISSUE TITLE
Adding freeze instruction in Bit Field

### ISSUE THREAD

**Issue by aqjune** [snu-sf/llvm-freeze#26]
Freeze instruction is added in C/C++ bit field (https://github.com/snu-sf/clang-freeze/pull/1).

I'll estimate performance degradation for this update. Also, I welcome any suggestions to test more C/C++ codes.

**Comment by nunoplopes** [comment]
Here it's important to measure the increase in memory usage and of compilation time as well.
I don't remember right now of any bit-field intensive benchmark.. Maybe you can grep over SPEC and llvm testsuite to see if there's anything there (e.g., grep ' [a-zA-Z0-9_]+:[0-9]+\s*;')

**Comment by aqjune** [comment]
I haven't thought about the memory usage..
To measure memory usage I'll try the method described in http://dl.acm.org/citation.cfm?id=1241618 . 
Also I'll estimate the compilation time as well.

**Comment by aqjune** [comment]
I identified 5 benchmarks in SPEC2006 that use bitfields.
Finding bitfield definition by regular expression was hard because ternary operator (? : ) also matched the pattern. I used other regular expressions, but it was hard for me to identify them. (http://stackoverflow.com/questions/16021715/regular-expression-to-find-all-structs-defined-with-a-bit-field)
So instead I used --save-temps option to clang and identified IR files which contain freeze instruction. As far as I understand clang with --save-temps outputs raw IR generated from AST, and the only case freeze instruction is generated is from assigning value into bitfield.

Here is the number of freeze-containing IR files as well as the number of of total IR files. It seems like gcc is using bitfield functionality intensively. soplex also uses bitfield a lot. soplex is a tool for solving linear program.

```
400.perlbench : 2/51
401.bzip2 : 0/9
403.gcc : 85/155
429.mcf : 0/11
433.milc : 0/68
444.namd : 1/11
445.gobmk : 0/66
447.dealII : 0/116
450.soplex : 30/63
453.povray : 3/100
456.hmmer : 0/57
458.sjeng : 0/19
462.libquantum : 0/16
464.h264ref : 0/42
470.lbm : 0/2
471.omnetpp : 0/85
473.astar : 0/11
482.sphinx3 : 0/44
483.xalancbmk : 0/738
```
